### ISSUE TITLE
Customized loss + metrics 

### ISSUE THREAD

**Issue by laurenf3395** [PhosphorylatedRabbits/brontes#19]
In the current implementation, the metrics and the loss can only be a function of the predicted logits and the class labels. This is enough when considering simple classification problems but won't work if we need more advanced losses/metrics (eg siamese loss, VAE loss).

Can we think of a more generic way of defining the losses/metrics in the Brontes constructor to cope with this limitation ?

**Comment by afoncubierta** [comment]
Addressed by @guillaumejaume  in #20
